feat: Provide Base EntityController (#96)
nestjs-addons/in-memory-db: README.md

This provides a great way to quickly get up and running with prototypes and mock backends.

> ## Table of Content

Reviewer: ```suggestion
## Table of Contents
```

Using this decorator ensures that the correct instance is injected.

## Entity Controller

> In order to prevent code duplication and boilerplate for each controller, we created two base entity controllers `InMemoryDbEntityController` and `InMemoryDbEntityAsyncController`.

Reviewer: ```suggestion
In order to prevent code duplication and boilerplate for each controller, we have created two base entity controllers `InMemoryDbEntityController` and `InMemoryDbEntityAsyncController`. This allows you to quickly provide endpoints to make requests without having to manually implement each action.
```

> To use the controllers, simply create a controller and extend it with one of the controllers.

Reviewer: ```suggestion
To use the controllers, simply create a new controller and extend it with one of the provided base controllers.
```

```typescript
@Controller('api/users')
class UsersController extends InMemoryDbController<UserEntity> {

  constructor(protected dbService: InMemoryDBService<UserEntity>) {
    super(dbService);
  }

}

```

> Of course, it will also be possible to use an Entity Controller with a specific feature service by using the decorator `InjectInMemoryDBService`

Reviewer: ```suggestion
In order to have an Entity Controller use a feature-specific instance of the service, use the decorator `InjectInMemoryDBService` in the controller's provided by this library as shown below:
```
